mne/time_frequency/psd.py

```diff
@@ -112,6 +114,202 @@ def _check_nfft(n, n_fft, n_overlap):
     return n_fft, n_overlap
 
 
+def _check_psd_data(inst, tmin, tmax, picks, proj):
+    """Helper to do checks on PSD data / pull arrays from inst"""
+    from ..io.base import _BaseRaw
+    from ..epochs import _BaseEpochs
+    from ..evoked import Evoked
+    if not isinstance(inst, (_BaseEpochs, _BaseRaw, Evoked)):
+        raise ValueError('epochs must be an instance of Epochs, Raw, or'
+                         'Evoked. Got type {0}'.format(type(inst)))
+
+    time_mask = _time_mask(inst.times, tmin, tmax)
+    if picks is None:
+        picks = pick_types(inst.info, meg=True, eeg=True, ref_meg=False,
+                           exclude='bads')
+    if proj:
+        inst = inst.apply_proj()
+
+    sfreq = inst.info['sfreq']
+    if isinstance(inst, _BaseRaw):
+        start, stop = np.where(time_mask)[0][[0, -1]]
+        data, times = inst[picks, start:(stop + 1)]
+    elif isinstance(inst, _BaseEpochs):
+        data = inst.get_data()[:, picks][..., time_mask]
+    elif isinstance(inst, Evoked):
+        data = inst.data[picks][..., time_mask]
+
+    return data, sfreq
+
+
+@verbose
```

Comment:
Hmmm, so I just removed the verbose kwargs from all these functions in lieu of having a decorator, but now I'm thinking maybe that's not the right answer? I see a lot of other functions that both use @verbose and also have a verbose kwarg